### ISSUE TITLE
Regarding feat: implement unthrottled concurrency using task queue

### ISSUE THREAD

**Issue by wumpus** [lc/gau#141]
Can you stop attacking the Common Crawl CDX API?

**Comment by lc** [comment]
I’m not? This is an open source tool to find archived URLs for a given domain…

**Comment by wumpus** [comment]
Yes, and because it isn't throttled, use of this package harms the target, which is me.

**Comment by wumpus** [comment]
Any progress? I was hoping for rate limiting, and honoring 503 and 429 status codes. And not just "unthrottled concurrency".

**Comment by lc** [comment]
It’s open source, you’re welcome to submit a PR.

**Comment by lc** [comment]
Accidentally closed

**Comment by wumpus** [comment]
Thanks for adding to your TODO list, I appreciate it!

Here's an example of making a single query in Athena that's much more efficient than gau: https://positive.security/blog/ransack-data-exfiltration#common-crawl

**Comment by lc** [comment]
Thanks for the reference & sorry about the slowness to implement. Getting hitched 😅

**Comment by wumpus** [comment]
Congratulations!

**Comment by mr-pmillz** [comment]
Any update on this?

**Comment by wumpus** [comment]
I have pretty vicious rate limits on the API now, so I expect that this software is broken.

**Comment by mr-pmillz** [comment]
Yeah i noticed that the commoncrawl module is borked in gau due to rate limits. 
gau still works for the other providers though.
Was curious the status of when gau was planned to be updated to add your suggested feature @wumpus
